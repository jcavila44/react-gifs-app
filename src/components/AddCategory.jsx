import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmitForm = (event) => {
        event.preventDefault();
        const newInputValueClean = inputValue.trim();
        if (newInputValueClean.length < 1) return;

        onNewCategory(newInputValueClean);
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmitForm} aria-label="form">
            <input
                key={12}
                type="text"
                placeholder="Buscar gifs..."
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
};

AddCategory.defaultProps = {
    onNewCategory: 'Faltan la funcion de onNewCategory'
};