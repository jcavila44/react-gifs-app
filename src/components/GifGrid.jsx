
import PropTypes from 'prop-types';
import { fetchGetGifs } from '../helpers/getGifs';
import { useEffect } from 'react';




export const GifGrid = ({ category }) => {

    useEffect(() => {
        fetchGetGifs(category);
    }, [category]);

    return (
        <>
            <h3>{category}</h3>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string
};

GifGrid.defaultProps = {
    category: 'Fata la propiedad category'
};