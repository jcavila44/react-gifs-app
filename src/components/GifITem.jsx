import PropTypes from 'prop-types';

export const GifITem = ({ title, url, }) => {
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}



GifITem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};

GifITem.defaultProps = {
    id: 'Fata la propiedad id',
    title: 'Fata la propiedad title',
    url: 'Fata la propiedad url',
};