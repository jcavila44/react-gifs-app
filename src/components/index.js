export * from './AddCategory' ;
export * from './GifGrid' ;
export * from './GifITem' ;