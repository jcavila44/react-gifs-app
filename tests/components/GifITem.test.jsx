import { render, screen } from "@testing-library/react"
import { GifITem } from "../../src/components/GifITem"

describe('Pruebas en <GifITem/>', () => {

    const title = 'Naruto';
    const url = 'https://naruto.com.co/naruto.jpg'
    const id = '123';

    test('Debe de hacer match con el snapshot ', () => {

        const { container } = render(<GifITem id={id} title={title} url={url} />);
        expect(container).toMatchSnapshot();

    })

    test('Debe de mostar la imagen con el URL y el ALT indicado', () => {

        render(<GifITem id={id} title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('should first', () => {

        render(<GifITem id={id} title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    
    })

})