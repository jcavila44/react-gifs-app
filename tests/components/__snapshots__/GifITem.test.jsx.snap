// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifITem/> Debe de hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://naruto.com.co/naruto.jpg"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
